projects/igniteui-angular/src/lib/grid/grid.search.spec.ts

```diff
@@ -1,715 +1,429 @@
+            jackSoftwareCell = grid.getCellByColumn(3, 'JobTitle').nativeElement;
+            activeHighlight = jackSoftwareCell.querySelector('.' + component.activeClass);
+            highlights = jackSoftwareCell.querySelectorAll('.' + component.highlightClass);
             expect(highlights.length).toBe(1);
             expect(activeHighlight).toBe(highlights[0]);
 
-            grid.deleteRow(3);
-            grid.cdr.detectChanges();
-
-            return fix.whenStable();
-        }).then(() => {
+            grid.deleteRow(4);
             fix.detectChanges();
-            celiaSoftwareCell = grid.getCellByColumn(3, 'JobTitle');
-            activeHighlight = celiaSoftwareCell.nativeElement.querySelector('.' + fix.componentInstance.activeClass);
-            highlights = celiaSoftwareCell.nativeElement.querySelectorAll('.' + fix.componentInstance.highlightClass);
+
+            celiaSoftwareCell = grid.getCellByColumn(3, 'JobTitle').nativeElement;
+            activeHighlight = celiaSoftwareCell.querySelector('.' + component.activeClass);
+            highlights = celiaSoftwareCell.querySelectorAll('.' + component.highlightClass);
             expect(highlights.length).toBe(1);
             expect(activeHighlight).toBe(highlights[0]);
 
             grid.findPrev('software');
 
-            return fix.whenStable();
-        }).then(() => {
             fix.detectChanges();
-            leslieSoftwareCell = grid.getCellByColumn(7, 'JobTitle');
-            activeHighlight = leslieSoftwareCell.nativeElement.querySelector('.' + fix.componentInstance.activeClass);
-            highlights = leslieSoftwareCell.nativeElement.querySelectorAll('.' + fix.componentInstance.highlightClass);
+            leslieSoftwareCell = grid.getCellByColumn(7, 'JobTitle').nativeElement;
+            activeHighlight = leslieSoftwareCell.querySelector('.' + component.activeClass);
+            highlights = leslieSoftwareCell.querySelectorAll('.' + component.highlightClass);
             expect(highlights.length).toBe(1);
             expect(activeHighlight).toBe(highlights[0]);
         });
-    }));
-
-    it('Highlights should be properly updated when a row is added', async(() => {
-        const fix = TestBed.createComponent(SimpleGridComponent);
-        fix.detectChanges();
-
-        const grid = fix.componentInstance.gridSearch;
-        const tanyaDirectorCell = grid.getCellByColumn(2, 'JobTitle');
-        let activeHighlight: any;
-        let highlights: any[];
 
-        grid.findNext('director');
-        grid.findNext('director');
+        it('Highlights should be properly updated when a row is added', () => {
+            const tanyaDirectorCell = grid.getCellByColumn(2, 'JobTitle').nativeElement;
+            let activeHighlight: any;
+            let highlights: any[];
 
-        fix.whenStable().then(() => {
+            grid.findNext('director');
+            grid.findNext('director');
```

Comment:
Is it necessary to call the `findNext()` two times?

Reply:
In this case yes, because the goal is to highlight the second occurence.